src/main/java/hudson/plugins/git/GitSCM.java

```diff
@@ -764,7 +764,9 @@ public GitClient createClient(TaskListener listener, EnvVars environment, Run<?,
                 StandardUsernameCredentials credentials = CredentialsMatchers.firstOrNull(urlCredentials, idMatcher);
                 if (credentials != null) {
                     c.addCredentials(url, credentials);
-                    CredentialsProvider.track(project.getLastBuild(), credentials);
+                    if (project != null && project.getLastBuild() != null) {
+                        CredentialsProvider.track(project.getLastBuild(), credentials);
```

Comment:
Tiny race condition risk (the one remaining build could be deleted between these two lines). Safer to assign the `lastBuild` to a local variable and check that for null.